nexus/db-queries/src/db/on_conflict_ext.rs

```diff
@@ -0,0 +1,36 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use diesel::dsl::sql;
+use diesel::expression::SqlLiteral;
+use diesel::prelude::*;
+use diesel::sql_types::Bool;
+use diesel::upsert::IncompleteOnConflict;
+
+/// An extension trait for incomplete [`ON CONFLICT`] expressions.
+pub trait IncompleteOnConflictExt {
+    type AsPartialIndexOutput;
+
+    // TODO: explain what this does.
```

Comment:
This is where I'd put in a pretty long comment.

Reply:
I'm interested. Happy to see that your issue to CRDB is going to get us more docs from them too! https://github.com/cockroachdb/docs/issues/18292 .

Reply:
Done -- see the latest version of the PR, and the screenshot of a rendered version of it.